src/librustc_codegen_ironox/type_.rs

```diff
@@ -158,11 +180,23 @@ impl BaseTypeMethods<'tcx> for CodegenCx<'ll, 'tcx> {
         els: &[Type],
         packed: bool
     ) -> Type {
-        unimplemented!("type_struct");
+        let mut types = self.types.borrow_mut();
+        let mut members = els.to_vec();
+        let struct_type = LLType::StructType {
+            name: None,
+            members,
+        };
+        types.push(struct_type);
+        types.len() - 1
     }
 
     fn type_array(&self, ty: Type, len: u64) -> Type {
-        unimplemented!("type_array");
+        let mut types = self.types.borrow_mut();
+        let array_type = LLType::Array {
```

Comment:
For consistency with the code below, I'd probably do `types.push(LLType::Array { ... })`.